fedimint-client/src/db/event_log.rs

```diff
@@ -336,13 +350,27 @@ pub(crate) async fn run_event_log_ordering_task(
                 dbtx.remove_entry(unordered_id).await.is_some(),
                 "Must never fail to remove entry"
             );
-            assert!(
-                dbtx.insert_entry(&next_entry_id, entry).await.is_none(),
-                "Must never overwrite existing event"
-            );
-
-            trace!(target: LOG_CLIENT_EVENT_LOG, ?unordered_id, id=?next_entry_id, "Ordered event log event");
-            next_entry_id = next_entry_id.next();
+            if entry.persist {
+                assert!(
+                    dbtx.insert_entry(&next_entry_id, &entry.inner)
+                        .await
+                        .is_none(),
+                    "Must never overwrite existing event"
+                );
```

Comment:
Hmm... This is running in it's own highest-level task spawned through `TaskGroup`, so it should surface correctly. Also, panic would cause events stop being processed altogether, so not something easy to miss, I think.